packages/blade/src/components/Carousel/types.ts

```diff
@@ -0,0 +1,87 @@
+import type { BoxProps } from '~components/Box';
+
+type OverlayColor = BoxProps['backgroundColor'];
+
+type CarouselProps = {
+  /**
+   * Accepts CarouselItem
+   */
+  children: React.ReactNode;
+  /**
+   * Total number of carousel items to show at once,
+   * if set to 1,2 or 3 all the CarouselItem's width will remain the same,
+   * but if set to `autofit` the carousel items will take up space responsively
+   *
+   * @default 1
+   */
+  visibleItems?: 1 | 2 | 3 | 'autofit';
+  /**
+   * If true, adds extra margin before and after the first/last slides so that they align in center
```

Comment:
Added note. 

We should not throw warnings because: 

Imagine in desktop we want it to have `shouldAddStartEndSpacing` and on mobile we automatically internally make this prop false based on the screen size. 

If we throw an warning consumers will unnecessary will have to add another isMobile check on their end just to remove this warning, even though we already handle it internally gracefully. 

```
const { deviceType } = useTheme();
const isMobile = deviceType === 'mobile'
<Carousel shouldAddStartEndSpacing={isMobile ? false : true} />
```